testing/object-storage-mock/src/main/java/org/projectnessie/objectstoragemock/HeapStorageBucket.java

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright (C) 2024 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.objectstoragemock;
+
+import static java.lang.System.currentTimeMillis;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.ConcurrentModificationException;
+import java.util.TreeMap;
+import java.util.stream.Stream;
+
+public class HeapStorageBucket {
+  private final TreeMap<String, MockObject> objects = new TreeMap<>();
+
+  private HeapStorageBucket() {}
+
+  public static HeapStorageBucket newHeapStorageBucket() {
```

Comment:
Not sure we need this static method + the `bucket()` method. I was rather expecting stg like:

```java
Bucket bucket = HeapStorageBucketFactory.createNew();
```

Reply:
Need `HeapStorageBucket` here, because `Bucket` is a rather algorithmic approach, while `HeapStorageBucket` has state.